build.sh

```diff
@@ -174,15 +175,6 @@ fi
 
 # Build and install the rmm Python package
 if (( NUMARGS == 0 )) || hasArg rmm; then
-    cd "${REPODIR}/python"
-    export INSTALL_PREFIX
-    echo "building rmm..."
-
-    python setup.py build_ext --inplace -- -DCMAKE_PREFIX_PATH=${INSTALL_PREFIX} ${EXTRA_CMAKE_ARGS}
-
-    if [[ ${INSTALL_TARGET} != "" ]]; then
-        echo "installing rmm..."
-        python setup.py install --single-version-externally-managed --record=record.txt -- -DCMAKE_PREFIX_PATH=${INSTALL_PREFIX} ${EXTRA_CMAKE_ARGS}
-    fi
-
+    echo "building and installing rmm..."
+    SKBUILD_CONFIGURE_OPTIONS="${SKBUILD_EXTRA_CMAKE_ARGS}" python -m pip install --no-build-isolation --no-deps ${REPODIR}/python
```

Comment:
Why `--no-deps`? As far as I can tell, the previous command would have installed missing dependencies.

Reply:
I'm not sure what you mean by the previous command. Do you mean the cmake build of librmm?

To be clear, `--no-deps` prevents the installation of _runtime_ dependencies, not build dependencies. This prevents us from e.g. installing numba (a runtime dependency of rmm) if we're executing build.sh in an environment where we just need to build it. We aren't guaranteed that all runtime dependencies are installed. This is less important for rmm than for other packages with much more extensive dependency lists, of course.

Reply:
By "previous command" I meant the `python setup.py build_ext --inplace -- ...`. Did that install runtime dependencies?

The behavior I am wondering about is whether users should expect to run `./build.sh` and then have an importable, usable package. That requires runtime dependencies. If that behavior has changed before/after this PR, then we should consider installing the runtime dependencies as part of `./build.sh`.

Reply:
Ah I see. `python setup.py build_ext --inplace` won't install dependencies, but `python setup.py install` would. However, I think we can safely assume that the target audience for build.sh is people who have satisfied both build and run dependencies prior to the build. I added the `--no-deps` because the most common use case is satisfying those dependencies with a conda environment, and in that case we don't want pip installing dependencies. IIRC this caused problems for cugraph (which did this migration ages ago) devs, which is why we added this flag there in the first place. I've also added in cudf and so far heard no complaints.

I agree that it's not entirely obvious what the exact mapping of old commands to new commands is, though. Matching the old way definitely requires turning off build isolation and it's clear that we don't expect build deps to be installed, but runtime dependencies are a bit blurrier. My decision is based more off of most common experiences.

Reply:
Well, darn. I should have seen that earlier, of course `build_ext` won’t install runtime dependencies. This is fine.